configuration.c

```diff
@@ -4592,6 +4600,36 @@ bool config_load_remap(const char *directory_input_remapping,
        || string_is_empty(directory_input_remapping))
       return false;
 
+   if (   sort_remaps_by_controller 
+       && input_device_name != NULL 
+       && !string_is_empty(input_device_name))
+   {
+      /* Ensure directory does not contain special chars */ 
+      input_device_dir = sanitize_path_part(input_device_name);
+      
+      /* Allocate memory for the new path */
+      remap_path_total_len = strlen(core_name) + strlen(input_device_dir) + 2;
+      remap_path = (char *)malloc(remap_path_total_len);
+
+      /*  Build the new path with the controller name */
+      strlcpy(remap_path, core_name, remap_path_total_len);
+      strlcat(remap_path, "/", remap_path_total_len);
```

Comment:
Can you concatenate using strlcpy here instead of strlcat? There are plenty of examples elsewhere in the codebase that shows how to do this - you can use a variable such as '_len' to keep track of the position/offset.

Reply:
Resolved in newest commit.